### DISCUSSION TITLE
Products' image storage

### DISCUSSION THREAD

**Discussion by emsitkowski** [medusajs/medusa#1993] [Q&A] [answered]
Hey,
I'm just getting started with Medusa, I find it revolutionary and so pleasant to use :)

The question that keeps me at night though is: 
Is there any alternative for hosting images? I'd love to try to host Medusa on Hostinger, and it'd be the best to keep everything there – including the images.

**Comment by abuuzayr** [comment, marked as answer]
Certainly, if Hostinger has an Image hosting service I suppose you can write your own plugin by extending the `AbstractFileService` class from `@medusajs/medusa` like in the S3 plugin: https://github.com/medusajs/medusa/blob/master/packages/medusa-file-s3/src/services/s3.js

Otherwise, if you are renting VPSes from Hostinger, you can always host a MinIO server on the same server or another standalone server to act as your own Image hosting service.

**Reply by emsitkowski** [reply]
Thank you, I’ll dig into this issue further and hopefully find some solution that I’m happy with :)

**Reply by moaazbhnas228** [reply]
Hi @emsitkowski ! Any luck with this?

**Reply by emsitkowski** [reply]
Hi, not really. I was just doing this for educational purposes and finally left this project. Good luck with that though!
